h2o-mapreduce-generic/src/main/java/water/hadoop/HiveTokenGenerator.java

```diff
@@ -47,20 +79,17 @@ private String getDelegationTokenFromConnection(String url, String principal, St
     }
   }
 
-  private void addHiveDelegationTokenIfPossible(Configuration conf, String hiveHost, String hivePrincipal, Credentials creds) throws IOException {
+  private void addHiveDelegationTokenIfPossible(
+      Configuration conf,
+      String hiveHost,
+      String hivePrincipal,
+      Credentials creds
+  ) throws IOException {
     if (!isHiveDriverPresent()) {
```

Comment:
Can we instead make sure this whole mechanism won't be triggered if the token renewal is not needed? It looks like we start the renewal threads regardless if we have Hive or not.